apis/configwatch_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Contributors to the L3AF Project.
+// SPDX-License-Identifier: Apache-2.0
+
+package apis
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMatchHostnamesWithRegexp(t *testing.T) {
+	type args struct {
+		pattern string
+		host    string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name:    "EmptypCheck",
+			args:    args{pattern: "", host: ""},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name:    "LengthMissMatchCheck",
+			args:    args{pattern: "l3afd-lfn.us.l3af.io", host: "l3afd-lfn.l3af.io"},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name:    "LengthMatchCheck",
+			args:    args{pattern: "l3afd-lfn.l3af.io", host: "l3afd-lfn.l3af.io"},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name:    "LengthMatchPatternMissCheck",
+			args:    args{pattern: "l3afd-us.l3af.io", host: "l3afd-lf.l3af.io"},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name:    "PatternMatchCheck",
+			args:    args{pattern: "l3afd-*.l3af.io", host: "l3afd-lfn.l3af.io"},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name:    "PatternMissMatchCheck",
+			args:    args{pattern: "*l3afd-.l3af.io", host: "l3afd-lfn.l3af.io"},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name:    "PatternRegExMatchCheck",
+			args:    args{pattern: "asnl3afd-lfn.l3af.io", host: ".*l3afd-lfn.l3af.io"},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name:    "PatternRegExExactMatchCheck",
+			args:    args{pattern: "l3afd-dev.l3af.io", host: "^dev.l3af.io$"},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name:    "PatternRegExFindMatch",
+			args:    args{pattern: "l3afd-dev.l3af.io", host: "dev.l3af.io"},
+			want:    true,
+			wantErr: false,
+		},
```

Comment:
I would suggest adding a regex that checks something like:

Pattern: "l3afd-dev-10.l3af.io"
host: "^l3afd-dev-[0-9][0-9]\.l3af\.io"
want: true
wantErr: false